ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoicePayloadExecutorTest.java

```diff
@@ -71,17 +82,36 @@ void setUp() {
   @Test
   void optimisticallyExecute_shouldSendToExecutionEngineAndReturnTrue() {
     final ForkChoicePayloadExecutor payloadExecutor = createPayloadExecutor();
-    final ExecutionPayload payload = dataStructureUtil.randomExecutionPayload();
-    final boolean result = payloadExecutor.optimisticallyExecute(payload);
+    final boolean result = payloadExecutor.optimisticallyExecute(payloadHeader, payload);
     verify(executionEngine).executePayload(payload);
     assertThat(result).isTrue();
   }
 
+  @Test
+  void optimisticallyExecute_shouldNotExecuteDefaultPayload() {
+    final ForkChoicePayloadExecutor payloadExecutor = createPayloadExecutor();
+    final boolean result = payloadExecutor.optimisticallyExecute(payloadHeader, defaultPayload);
+    verify(executionEngine, never()).executePayload(any());
+    assertThat(result).isTrue();
+  }
+
+  @Test
+  void optimisticallyExecute_shouldValidateMergeBlockWhenThisIsTheMergeBlock() {
```

Comment:
this seems more like `optimisticallyExecute_shouldCheckParentHashBeforExecutingPayload` or something?